core/chain-configs/src/genesis_config.rs

```diff
@@ -234,20 +234,57 @@ impl From<&GenesisConfig> for AllEpochConfig {
 )]
 pub struct GenesisRecords(pub Vec<StateRecord>);
 
+/// custom deserializer that does the same thing that #[serde(default)] would do --
+/// if no value is provided in json, returns default value.
+fn no_value_as_default<'de, D, T>(de: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de> + Default,
+{
+    let opt = Option::<T>::deserialize(de)?;
```

Comment:
I believe using an `Option` here allows `null` to become the default value as  well, whereas the usual `#[serde(default)]` approaches don't permit that. Up to you if that's the desired behaviour or not.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2f94de6bae1c3c86b0febeae4da4960a

Reply:
I think I am Ok with `null` case, but it is better to be clear about that.